core/chains/evm/txm/txm.go

```diff
@@ -0,0 +1,462 @@
+		Name: "txm_num_confirmed_transactions",
+		Help: "Total number of confirmed transactions. Note that this can happen multiple times per transaction in the case of re-orgs or when filling the nonce for untracked transactions.",
+	}, []string{"chainID"})
+	promNumNonceGaps = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "txm_num_nonce_gaps",
+		Help: "Total number of nonce gaps created that the transaction manager had to fill.",
+	}, []string{"chainID"})
+	promTimeUntilTxConfirmed = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "txm_time_until_tx_confirmed",
+		Help: "The amount of time elapsed from a transaction being broadcast to being included in a block.",
+	}, []string{"chainID"})
+)
+
+type Config struct {
+	EIP1559             bool
+	BlockTime           time.Duration
+	RetryBlockThreshold uint16
+	EmptyTxLimitDefault uint64
+}
+
+type Txm struct {
+	services.StateMachine
+	lggr            logger.SugaredLogger
+	chainID         *big.Int
+	client          Client
+	attemptBuilder  AttemptBuilder
+	errorHandler    ErrorHandler
+	stuckTxDetector StuckTxDetector
+	txStore         TxStore
+	keystore        Keystore
+	config          Config
+
+	nonceMapMu sync.Mutex
+	nonceMap   map[common.Address]uint64
+
+	triggerCh map[common.Address]chan struct{}
+	stopCh    services.StopChan
+	wg        sync.WaitGroup
+}
+
+func NewTxm(lggr logger.Logger, chainID *big.Int, client Client, attemptBuilder AttemptBuilder, txStore TxStore, stuckTxDetector StuckTxDetector, config Config, keystore Keystore) *Txm {
+	return &Txm{
+		lggr:            logger.Sugared(logger.Named(lggr, "Txm")),
+		keystore:        keystore,
+		chainID:         chainID,
+		client:          client,
+		attemptBuilder:  attemptBuilder,
+		txStore:         txStore,
+		stuckTxDetector: stuckTxDetector,
+		config:          config,
+		nonceMap:        make(map[common.Address]uint64),
+		triggerCh:       make(map[common.Address]chan struct{}),
+	}
+}
+
+func (t *Txm) Start(ctx context.Context) error {
+	return t.StartOnce("Txm", func() error {
+		t.stopCh = make(chan struct{})
+
+		addresses, err := t.keystore.EnabledAddressesForChain(ctx, t.chainID)
+		if err != nil {
+			return err
+		}
+		for _, address := range addresses {
+			err := t.startAddress(ctx, address)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+}
+
+func (t *Txm) startAddress(ctx context.Context, address common.Address) error {
+	triggerCh := make(chan struct{}, 1)
+	t.triggerCh[address] = triggerCh
+	pendingNonce, err := t.pollForPendingNonce(ctx, address)
+	if err != nil {
+		return err
+	}
+	t.setNonce(address, pendingNonce)
+
+	t.wg.Add(2)
+	go t.broadcastLoop(address, triggerCh)
+	go t.backfillLoop(address)
+	return nil
+}
+
+func (t *Txm) pollForPendingNonce(ctx context.Context, address common.Address) (pendingNonce uint64, err error) {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, pendingNonceDefaultTimeout)
+	defer cancel()
+	for {
```

Comment:
This will block the node startup if the RPC isn't returning success for some reason.